Add `DROP DATABASE name CASCADE ALIASES` (#1805)
neo4j/docs-operations: modules/ROOT/pages/database-administration/standard-databases/delete-databases.adoc

DROP DATABASE customers IF EXISTS DUMP DATA
----

[[delete-databases-with-aliases]]
=== Delete a database with local database aliases targeting it

In general, deleting a database that is the target of any local database aliases will fail due to those database aliases.
Any remote database aliases targeting the database does not affect the deletion of the database.

By appending `CASCADE ALIASES` to the command `DROP DATABASE`, those local database aliases can be dropped together with the database.

> Any remote database aliases targeting the database will not be affected of the `CASCADE ALIASES`, they will simply no longer resolve their targets (which is the same behaviour as if you create them targeting a non-existing database).

Reviewer: Doesn't do any harm, although I don't think I would assume it would.

DROP DATABASE customers IF EXISTS DUMP DATA
----

[[delete-databases-with-aliases]]
=== Delete a database with local database aliases targeting it

> In general, deleting a database that is the target of any local database aliases will fail due to those database aliases.

Reviewer: ```suggestion
In general, deleting a database that is the target of any local database aliases will fail due as these aliases need to be dropped first.
```

> Any remote database aliases targeting the database does not affect the deletion of the database.

Reviewer: ```suggestion
Remote database aliases targeting the database do not affect the deletion of the database.
```

> By appending `CASCADE ALIASES` to the command `DROP DATABASE`, those local database aliases can be dropped together with the database.

Reviewer: ```suggestion
By appending `CASCADE ALIASES` to the command `DROP DATABASE`, those local database aliases can be dropped alongside the database.
```

> Any remote database aliases targeting the database will not be affected of the `CASCADE ALIASES`, they will simply no longer resolve their targets (which is the same behaviour as if you create them targeting a non-existing database).

Reviewer: ```suggestion
Any remote database aliases targeting the database will not be affected of the `CASCADE ALIASES`, they will simply no longer resolve their targets (which is the same behaviour as if they are created targeting a non-existing database).
```
Again US / UK spelling of behaviour, docs team probably can confirm.

Author: ```suggestion
Any remote database aliases targeting the database will not be affected of the `CASCADE ALIASES`, they will simply no longer resolve their targets (which is the same behavior as if they are created targeting a non-existing database).
```

Using `CASCADE ALIASES` requires the `DROP ALIAS` privilege (see xref:authentication-authorization/dbms-administration.adoc#access-control-dbms-administration-alias-management[ALIAS MANAGEMENT privileges]).

Given database and database alias:
[source, cypher]
----
CREATE DATABASE movies
CREATE ALIAS films FOR DATABASE movies
----

Then both the database alias `films` and the database `movies` will be dropped with:
[source, cypher]
----
DROP DATABASE movies CASCADE ALIASES
----

The option `RESTRICT` explicitly requests the default behavior of the command.

[NOTE]
====

> The `CASCADE ALIASES` option will drop the same aliases that would be given in the error message from trying to drop the database using the `RESTRICT` mode instead.

Reviewer: As above, I don't think we should mention the error message.

DROP DATABASE customers IF EXISTS DUMP DATA
----

[[delete-databases-with-aliases]]
=== Delete a database with local database aliases targeting it

In general, deleting a database that is the target of any local database aliases will fail due as these aliases need to be dropped first.
Remote database aliases targeting the database do not affect the deletion of the database.

By appending `CASCADE ALIASES` to the command `DROP DATABASE`, those local database aliases can be dropped alongside the database.
Any remote database aliases targeting the database will not be affected of the `CASCADE ALIASES`, they will simply no longer resolve their targets (which is the same behavior as if they are created targeting a non-existing database).
However, the behavior does not cascade further, if any of the dropped database aliases are constituents to any composite database, those composite databases are not dropped.

Using `CASCADE ALIASES` requires the `DROP ALIAS` privilege (see xref:authentication-authorization/dbms-administration.adoc#access-control-dbms-administration-alias-management[ALIAS MANAGEMENT privileges]).

> Given database and database alias:

Author: wouldn't this and the next bit make them into two separate examples? this is more set-up for the drop part and feels odd to have as an example on its own here?

> Given database and database alias:

Reviewer 2: Actually, I would suggest to be one example within an example framework. But I need to build it locally to see how it will look. I find it difficult to see the result from just commenting.

> Given database and database alias:

Reviewer 2: I'll suggest something.

> Given database and database alias:

Reviewer 2: We can have a similar example and wording for the other example. But I think we should see if we can reorganize that page as well, as at the moment, it discusses two topics—one is in the intro (_You can delete composite databases using either the command DROP COMPOSITE DATABASE name or the more general one DROP DATABASE name_), and the other is the new section, which is another way of dropping a composite database.

* Drop the local database aliases first, then use `DROP DATABASE name` to drop the database.
Remote database aliases targeting the database do not affect the deletion of the database.
* Use `DROP DATABASE name CASCADE ALIASES` to also drop the local database aliases targeting it while dropping the database.
If any of the dropped database aliases are constituents of composite databases, those composite databases will not be dropped.
This command does not affect the remote database aliases targeting the database being dropped.
They will simply no longer resolve their targets as if they were created targeting a non-existing database.

Using `CASCADE ALIASES` requires the `DROP ALIAS` privilege.
For more information, see xref:authentication-authorization/dbms-administration.adoc#access-control-dbms-administration-alias-management[ALIAS MANAGEMENT privileges].

Given database and database alias:
[source, cypher]
----
CREATE DATABASE movies
CREATE ALIAS films FOR DATABASE movies
----

Then both the database alias `films` and the database `movies` will be dropped with:
[source, cypher]
----
DROP DATABASE movies CASCADE ALIASES
----

> 

Reviewer 2: ```suggestion
====
```

DROP DATABASE customers IF EXISTS DUMP DATA
----

[[delete-databases-with-aliases]]
=== Delete a database with local database aliases targeting it

There are two ways of dropping a database that is the target of local database aliases:

* Drop the local database aliases first, then use `DROP DATABASE name` to drop the database.

> Remote database aliases targeting the database do not affect the deletion of the database.

Author: ```suggestion
Remote database aliases targeting the database do not affect the deletion of the database and therefore dos not need to be dropped beforehand.
```

* Use `DROP DATABASE name CASCADE ALIASES` to also drop the local database aliases targeting it while dropping the database.
If any of the dropped database aliases are constituents of composite databases, those composite databases will not be dropped.
This command does not affect the remote database aliases targeting the database being dropped.
They will simply no longer resolve their targets as if they were created targeting a non-existing database.

Using `CASCADE ALIASES` requires the `DROP ALIAS` privilege.
For more information, see xref:authentication-authorization/dbms-administration.adoc#access-control-dbms-administration-alias-management[ALIAS MANAGEMENT privileges].

.Drop a database and the local database alias targeting it
====
.Delete a database with local database aliases targeting it
====

> The following example assumes that you have a database that is the target of a local database alias:

Reviewer 2: ![Uploading Screenshot 2024-09-24 at 15.36.25.png…]()

> The following example assumes that you have a database that is the target of a local database alias:

Author: If I go to the current built version in the auto PR comment below it's still like this

> The following example assumes that you have a database that is the target of a local database alias:

Reviewer 2: Hmm, this is odd. It doesn't look like this locally.

> The following example assumes that you have a database that is the target of a local database alias:

Author: that would do it XD

> .Drop a database and the local database alias targeting it
> ====
> .Delete a database with local database aliases targeting it
> ====
> The following example assumes that you have a database that is the target of a local database alias:

Author: ```suggestion
.Drop a database and the local database alias targeting it
====
The following example assumes that you have a database that is the target of a local database alias:
```

[source, cypher]
----
CREATE DATABASE movies
CREATE ALIAS films FOR DATABASE movies
----

Then, both the database alias `films` and the database `movies` can be dropped with the following:
[source, cypher]
----
DROP DATABASE movies CASCADE ALIASES
----

> ====
> ====

Author: ```suggestion
====
```

> //The option `RESTRICT` explicitly requests the default behavior of the command.

Reviewer 2: ```suggestion
The option `RESTRICT` explicitly requests the default behavior of the command.
```

.Drop a database and the local database alias targeting it
====
The following example creates a database `movies` and a local database alias `films` targeting it:

[source, cypher]
----
CREATE DATABASE movies
CREATE ALIAS films FOR DATABASE movies
----

Then, the database `movies` and the local database alias `films` can be dropped using the following command:

[source, cypher]
----
DROP DATABASE movies CASCADE ALIASES
----

====

> //The option `RESTRICT` explicitly requests the default behavior of the command.

Author: Here it feels more like we should mention it somewhere, as this page have the syntax description at the top... my comment was on the composite page
